ShowBuildMenu.bat

```diff
@@ -165,7 +165,9 @@ echo Configuration activated.
 goto main-menu
 
 :test-run
-start "nunit3-console" cmd /K %NUNIT% --x86 --agents=1 --process=separate NHibernate.nunit
+SET NUNITPLATFORM=
+IF /I "%PLATFORM%" NEQ "x64" set NUNITPLATFORM=--x86
+start "nunit3-console" cmd /K %NUNIT% %NUNITPLATFORM% --agents=1 --process=separate NHibernate.nunit
```

Comment:
%PLATFORM% environment variable looks harmful, it causes build failures due to MsBuild using it if defined, as in [here](https://github.com/fsharp/FAKE/issues/1312#issuecomment-234196381).

It seems anyway that its default is to be not defined, it is just unfortunately set by this `ShowBuildMenu` when setting up test configuration, causing then build failures (especially on async file generation).

So I wonder if this change works as expected. I think in most cases we are still running nunit as a 32 bits process, even on x64 boxes. Maybe should be changed to use a logic like [here](https://stackoverflow.com/a/11383939/1178314).

Reply:
NUnit does run in x64 with this change...

The `PLATFORM` environment variable may be interacting with the async generation elsewhere, but it's not 'harmful' on this line in the `ShowBuildMenu.bat`.  This pull request didn't set the `PLATFORM`, just tested its value.

Reply:
> NUnit does run in x64 with this change...

Maybe on your machine, not on mine. This `PLATFORM` environment variable is not a standard thing. Most boxes do not have it. So your change may mostly works on your machine, not elsewhere. That is why I have link an alternate way using environment variables documented for this usage. (No official doc, but at least many references, like [here](https://blogs.msdn.microsoft.com/david.wang/2006/03/27/howto-detect-process-bitness/) and other places.)

Reply:
The point of this wasn't for general use x64 unit testing, it was only for enabling the testing of the x64 configurations in the build menu (of which there are ones for Firebird, SQLite, and SQL Server Compact).  I agree that even at this task it's imperfect as the configurations needs to be setup and then the tests ran in the same session of `ShowBuildMenu.bat`.

It's not about my machine or your machine, as none of my machines ever have `PLATFORM` set at a machine or user profile level.  It was intended to be entirely within the `ShowBuildMenu.bat`.  `PLATFORM` has been set in the `ShowBuildMenu.bat` since 2011:

https://github.com/nhibernate/nhibernate-core/blob/dfb877fa3cb2655f35aed182c0fa47aff307f7e5/ShowBuildMenu.bat#L98

Maybe a better solution would probably have two separate `Run tests using active configuration` one for each of x64 and x86.

The best solution would use NuGet packages of the ADO.NET drivers where both sets of native binaries are included in the build process and don't have to be selected at the configuration level, but based on the word size of the currently running process.

Reply:
Well it is worst than imperfect, since it is a "work only once" solution. Usually we do not re-create configurations for each test run, we just activate the one we want. And activation is not setting this variable.
By the way it cannot work when creating a SQL Server Compact test configuration either: this one put `AnyCpu` in this variable whatever its bitness.

So yes I agree that two options is the way to go. But that will require a letter shift, like moving B&C to A&B.

About NuGet, see #1401.

Firebird and SQLite data providers are CPU agnostic nowadays anyway. SQL Server CE is still an issue and this one will likely be solved only by dropping it. Oracle stays a special case.

Reply:
I was reacting to the comment that this pull request was somehow 'harmful', not that it only barely helps/works.

I think that adding to your #1401 pull request separate menu options to run tests in both x86 and x64 is a good idea.